torch/csrc/jit/serialization/import_read.cpp

```diff
@@ -29,10 +29,33 @@ IValue readArchiveAndTensors(
     return len;
   };
 
-  std::string archive_name_plus_slash = archive_name + "/";
+  static const char slash = '/';
   auto read_record = [&](const std::string& name) {
-    std::string ss = archive_name_plus_slash + name;
-    return std::get<0>(stream_reader.getRecord(ss));
+    std::size_t found = name.find(slash);
+    std::stringstream ss;
+    // In bytecode version 4, the tensor root_key doesn't include the parent
```

Comment:
Is this logic here because we don't want to require a backport to v4?
don't we anyway will need backport logic once we bump to v5 since there's other changes beyond the / ?

Reply:
Oh actually this logic belongs to #56002 (reuse constant table), because it's not related to get bytecode model version. It's a bit messed up when I restack the changes. This is for backward compatibility. It means that client running v5 can load model v4.